src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1442,6 +1443,13 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
                         "ATTACH ... FROM ... query is not supported for {} table engine, "
                         "because such tables do not store any data on disk. Use CREATE instead.", res->getName());
 
+    if (getContext()->getSettingsRef().create_replicated_merge_tree_fault_injection)
```

Comment:
Ah, initially I was wondering why we skip the `database->createTable` when the fault injection is enabled. Checking code, `StorageReplicatedMergeTree` constructor contains the invocation of `StorageReplicatedMergeTree::createReplica` (the one we add ZK node creation check), and `database->createTable` is just to make add entries/registry of the table to the Database part.